generators/import-jdl/index.js

```diff
@@ -92,77 +92,159 @@ module.exports = class extends BaseGenerator {
-                        document: jhiCore.parseFromFiles(this.jdlFiles),
+                    const parsed = jhiCore.parseFromFiles(this.jdlFiles);
+                    this.jdlObject = jhiCore.convertToJDLFromConfigurationObject({
+                        document: parsed,
                         databaseType: this.prodDatabaseType,
                         applicationType: this.applicationType,
                         applicationName: this.baseName
                     });
+                    if (shouldGenerateApplications(this.jdlObject)) {
+                        this.log('Writing application configuration files.');
+                        this.jdlObject.applications =
+                            setGeneratorVersionInJDLApplications(this.jdlObject.applications, packagejs.version);
+                        jhiCore.exportApplications({
+                            applications: this.jdlObject.applications,
+                            paths: getApplicationPaths(this.jdlObject.applications)
+                        });
+                    }
                     const entities = jhiCore.convertToJHipsterJSON({
-                        jdlObject,
+                        jdlObject: this.jdlObject,
                         databaseType: this.prodDatabaseType,
                         applicationType: this.applicationType
                     });
                     this.log('Writing entity JSON files.');
-                    this.changedEntities = jhiCore.exportEntities({
-                        entities,
-                        forceNoFiltering: this.options.force,
-                        applicationType: this.applicationType,
-                        applicationName: this.baseName
-                    });
-                    this.updatedKeys = Object.keys(this.changedEntities);
-                    if (this.updatedKeys.length > 0) {
-                        this.log(`Updated entities are: ${chalk.yellow(this.updatedKeys)}`);
+                    if (shouldGenerateApplications(this.jdlObject)) {
+                        this.changedEntities = jhiCore.exportEntitiesInApplications({
+                            entities,
+                            forceNoFiltering: this.options.force,
+                            applications: this.jdlObject.applications
+                        }).toArray();
                     } else {
-                        this.log(chalk.yellow('No change in entity configurations. No entities were updated'));
+                        this.changedEntities = jhiCore.exportEntities({
+                            entities,
+                            forceNoFiltering: this.options.force,
+                            application: {
+                                type: this.applicationType,
+                                name: this.baseName
+                            }
+                        });
                     }
-                } catch (e) {
-                    this.debug('Error:', e);
```

Comment:
this was added to help errors be more understandable to end users, `this.error(`Error while parsing applications and entities from the JDL ${error}.`);` is not very helpful all the time, so can you retain these, please

Reply:
Will do